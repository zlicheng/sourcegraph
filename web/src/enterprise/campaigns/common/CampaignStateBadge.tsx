import React from 'react'
import * as GQL from '../../../../../shared/src/graphql/schema'
import { CampaignsIcon } from '../icons'

interface Props {
    campaign: Pick<GQL.ICampaign, 'closedAt'>
    className?: string
}

export const CampaignStateBadge: React.FunctionComponent<Props> = ({ campaign, className = '' }) => {
    const isClosed = Boolean(campaign.closedAt)
    const badgeClassName = isClosed ? 'badge-danger' : 'badge-success'
    const badgeLabel = isClosed ? 'Closed' : 'Open'
    return (
        <span
            className={`badge ${badgeClassName} d-inline-flex align-items-center ${className} pr-2`}
            style={{ fontSize: 'unset' }}
        >
            <CampaignsIcon className="icon-inline mr-1" /> {badgeLabel}
        </span>
    )
}
