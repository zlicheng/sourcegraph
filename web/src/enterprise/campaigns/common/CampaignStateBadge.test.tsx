import React from 'react'
import { createRenderer } from 'react-test-renderer/shallow'
import { CampaignStateBadge } from './CampaignStateBadge'
import * as H from 'history'

describe('CampaignStateBadge', () => {
    test('', () =>
        expect(
            createRenderer().render(
                <CampaignStateBadge
                    campaign={{
                        author: { avatarURL: null, username: 'alice' },
                        createdAt: '2020-01-01',
                        description: '**a** b',
                    }}
                    history={H.createMemoryHistory()}
                />
            )
        ).toMatchSnapshot())
})
