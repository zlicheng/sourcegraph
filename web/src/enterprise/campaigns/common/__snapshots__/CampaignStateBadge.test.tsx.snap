// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CampaignStateBadge  1`] = `
<span
  className="badge badge-success mr-2"
>
  <Memo(ImageAutoAdjustIcon)
    className="icon-inline"
  />
   
  Open
</span>
`;
